tests/gen_many_js_functions.py

```diff
@@ -0,0 +1,40 @@
+import sys
+
+NUM_FUNCS_TO_GENERATE = 1000
+
+def func_name(i):
+	return 'thisIsAFunctionWithVeryLongFunctionNameThatWouldBeGreatToBeMinifiedWhenImportingToAsmJsOrWasmSideCodeToCallOtherwiseCodeSizesWillBeLargeAndNetworkTransfersBecomeVerySlowThatUsersWillGoAwayAndVisitSomeOtherSiteInsteadAndThenWebAssemblyDeveloperIsSadOrEvenWorseNobodyNoticesButInternetPipesWillGetMoreCongestedWhichContributesToGlobalWarmingAndThenEveryoneElseWillBeSadAsWellEspeciallyThePolarBearsAndPenguinsJustThinkAboutThePenguins' + str(i+1)
+
+def generate_js_library_with_lots_of_functions(out_file):
+	f = open(out_file, 'w')
+
+	f.write('var FunctionsLibrary = {\n')
+
+	for i in range(NUM_FUNCS_TO_GENERATE):
+		f.write('  ' + func_name(i) + ': function() { return ' + str(i+1) + '; },\n')
+
+	f.write('}\n');
+	f.write('mergeInto(LibraryManager.library, FunctionsLibrary);\n');
+	f.close()
+
+def generate_c_program_that_calls_js_library_with_lots_of_functions(out_file):
+	f = open(out_file, 'w')
+
+	f.write('#include <stdio.h>\n\n')
+
+	for i in range(NUM_FUNCS_TO_GENERATE):
+		f.write('int ' + func_name(i) + '(void);\n')
+
+	f.write('\nint main() {\n')
+	f.write('  int sum = 0;\n')
+
+	for i in range(NUM_FUNCS_TO_GENERATE):
+		f.write('  sum += ' + func_name(i) + '();\n')
+
+	f.write('\n  printf("Sum of numbers from 1 to ' + str(NUM_FUNCS_TO_GENERATE) + ': %d (expected ' + str((NUM_FUNCS_TO_GENERATE * (NUM_FUNCS_TO_GENERATE+1))/2) + ')\\n", sum);\n');
```

Comment:
I was expecting to see this code get run somewhere, but it looks like the test just compiles it never runs it.  Is that intentional?

Reply:
Thanks, added a run.